docs/serving/samples/grpc-ping-go/Dockerfile

```diff
@@ -12,21 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang AS builder
+# Use the offical Golang image to create a build artifact.
+# This is based on Debian and sets the GOPATH to /go.
+# https://hub.docker.com/_/golang
+FROM golang:1.12 as builder
 
-# Get the dependencies from GitHub
+# Retrieve the dependencies.
 RUN go get google.golang.org/grpc
 
-WORKDIR /go/src/github.com/knative/docs/
-ADD . /go/src/github.com/knative/docs/
+# Copy local code to the container image.
+WORKDIR /go/src/github.com/knative/docs/docs/serving/samples/grpc-ping-go
+COPY . ./
 
-RUN CGO_ENABLED=0 go build -tags=grpcping ./docs/serving/samples/grpc-ping-go/
-RUN CGO_ENABLED=0 go build -tags=grpcping ./docs/serving/samples/grpc-ping-go/client
+# Build the command inside the container.
+# To facilitate gRPC testing, this container includes a client command.
+RUN CGO_ENABLED=0 go build -tags=grpcping -o ./ping-server
+RUN CGO_ENABLED=0 go build -tags=grpcping -o ./ping-client ./client
```

Comment:
Where do the folders, ping-server and ping-client, come from?

Reply:
Those are the binary names generated by the build command. The first is building main in the current directory, the second is building the client directory.

Reply:
Thanks